### PR TITLE
Refactor `activerecord/Rakefile` and add `port` option to the config

### PR DESCRIPTION
**Pull request by davidstosik**
### Motivation / Background

My first need was to change the port used to connect to my local MySQL server when running commands like `bundle exec rake db:mysql:build`.

As I worked my way through that, I realized I could refactor make the code in the Rakefile DRYer, which is the second part of this PR.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/Rakefile [hunk 1]
```diff
@@ -207,52 +207,50 @@ end
 
 namespace :db do
   namespace :mysql do
-    mysql2_config = ARTest.config["connections"]["mysql2"]
-    mysql2_connection_arguments = lambda do |connection_name|
-      mysql2_connection = mysql2_config[connection_name]
-      ["--user=#{mysql2_connection["username"]}", ("--password=#{mysql2_connection["password"]}" if mysql2_connection["password"]), ("--host=#{mysql2_connection["host"]}" if mysql2_connection["host"]), ("--socket=#{mysql2_connection["socket"]}" if mysql2_connection["socket"])].join(" ")
+    mysql_configs = ARTest.config["connections"].slice("mysql2", "trilogy").transform_values { |connections| connections.slice("arunit", "arunit2") }
```
Comment (gmcgibbon) [inline activerecord/Rakefile:210]: Looks like `connector` and `connection_name` aren't used in your iterations below, so maybe we could just iterate the connection configs like before?

Comment (davidstosik) [reply activerecord/Rakefile:210]: The connection config includes more keys than `mysql2` and `trilogy` (`sqlite3` and `postgresql`) at the top level, and more keys than `arunit` and `arunit2` in the sub-level, stuff generated here:
https://github.com/rails/rails/blob/1717c0ef058c17c80e6e8d1bae14d6d18c8ea64b/activerecord/test/support/config.rb#L28-L29

These `slice` calls reduce the content of `mysql_configs` to what we need to setup the MySQL users and databases.

File: activerecord/Rakefile [hunk 2]
```diff
@@ -207,52 +207,50 @@ end
...
-        mysql_command = "mysql -uroot -proot -e"
-      else
-        mysql_command = "mysql -uroot -e"
-      end
-
-      mysql_configs.each do |config|
-        %x( #{mysql_command} "CREATE USER IF NOT EXISTS '#{config["arunit"]["username"]}'@'%';" )
-        %x( #{mysql_command} "CREATE USER IF NOT EXISTS '#{config["arunit2"]["username"]}'@'%';" )
-        %x( #{mysql_command} "GRANT ALL PRIVILEGES ON #{config["arunit"]["database"]}.* to '#{config["arunit"]["username"]}'@'%'" )
-        %x( #{mysql_command} "GRANT ALL PRIVILEGES ON #{config["arunit2"]["database"]}.* to '#{config["arunit2"]["username"]}'@'%'" )
```
Comment (davidstosik) [inline activerecord/Rakefile:236]: @gmcgibbon Note how these 4 lines are basically two copy-pasted pairs using a different connection name (`arunit` and `arunit2`).
Iterating through them means I don't have to repeat the MySQL command and the connection name.

File: activerecord/Rakefile (hunk 1 above)
Comment (gmcgibbon) [reply activerecord/Rakefile:210]:
```
mysql_configs = ARTest.config["connections"].values_at("mysql2", "trilogy").flat_map do |connections|
  connections.values_at("arunit", "arunit2")
end
```
Is what I had in mind, yes. Less noise in the loops.

File: activerecord/Rakefile [hunk 3]
```diff
@@ -207,52 +207,50 @@ end
...
+      ].join(" ")
     end
 
-    trilogy_config = ARTest.config["connections"]["trilogy"]
-    trilogy_connection_arguments = lambda do |connection_name|
-      trilogy_connection = trilogy_config[connection_name]
-      ["--user=#{trilogy_connection["username"]}", ("--password=#{trilogy_connection["password"]}" if trilogy_connection["password"]), ("--host=#{trilogy_connection["host"]}" if trilogy_connection["host"]), ("--socket=#{trilogy_connection["socket"]}" if trilogy_connection["socket"])].join(" ")
-    end
-
-    mysql_configs = [mysql2_config, trilogy_config]
```
Comment (davidstosik) [inline activerecord/Rakefile:222]: `mysql2_config` and `trilogy_config` are hashes which keys contain `arunit`, `arunit2`, and more other keys.

File: activerecord/Rakefile (hunk 1 above)
Comment (davidstosik) [reply activerecord/Rakefile:210]: Will update. 👍🏻

**Pull request merged**